packages/block-library/src/paragraph/edit.js

```diff
@@ -131,6 +147,7 @@ function ParagraphBlock( {
 						value={ fontSize.size }
 						onChange={ setFontSize }
 					/>
+					<LineHeightControl />
```

Comment:
The way to use the control is nice. One does not need to pass anything, but it is different from the way we use all the other controls. I would expect that in a control component, I would be able to pass a value, and onChange so I could store the data in the way I want or do something when the value is changed.
Would it make sense to support both usages e.g.: we use attributes and the way things work right now when value and onChange are not passed, but we still allow value and onChange?
I guess as a follow-up, we may see if other controls we have right now could also include support for directly using attributes.

Reply:
As part of having these implicit attributes, I'd love to see if we can reduce the need to handle attribute changes.

Reply:
Interesting point! Maybe the naming of the component would be different as it's API is different compared with our other controls.

The idea for the new `LineHeightControl` is to make it easy to plop in. As you noted, it's nice how simple it is 😊 .

We technically could support both use cases. Although... I'm not sure if we need (thinking out loud). The control itself doesn't really do much. It just connects the set + setState mechanisms with a `TextControl` primitive.

If a user needed to do this, they could replicate the same thing.
For more complex controls (e.g. the [UnitControl](https://github.com/WordPress/gutenberg/pull/20888) I'm adding for Cover), we can create a special primitive as part of `@wordpress/components`, and have the hooked up component within `@wordpress/block-editor`.

One primitive for customized implementation.
One preconfigured for a prop-free integration

Reply:
I agree a new naming convention would make sense here. It also goes back to my earlier concerns where I would expect a `*Control` to have the `onChange` and `value` props, and was concerned about some of the magic. The naming convention has a lot to do with that expectation.

I don't have a better idea for the naming convention but I do think that's a big part of it.

Reply:
> The naming convention has a lot to do with that expectation.

@chrisvanpatten That's a fair point. In that case... what do folks think about:

If *no props passed*... `<LineHeightControl />` would implicitly use values from `attributes`.

Otherwise, if `value` and `onChange` props are passed, `<LineHeightControl />` would use those instead.

Reply:
> If no props passed... <LineHeightControl /> would implicitly use values from attributes.

> Otherwise, if value and onChange props are passed, <LineHeightControl /> would use those instead.

This seems like a good approach 👍

Reply:
For me and since we're going to try an approach like #21021 where block authors won't have to use that component directly and would just enable a flag, I feel the magic/implicit approach won't be needed anymore. The implicitness will come from the hooks, not from the component itself.

I might take some time today to try to see how to apply the colors approach to the line-height feature here.

Reply:
Awesome! Interested to see how that goes! 🙏

Reply:
I implemented this in 6c7dcbaa94 

Here are my takeaways:

 - I like the consistency with other styles (color for now explored in #21037) and the simplicity.  
 - It's neat to be able to define support for a customization with a single line `__experimentalLineHeight: true`
 - It's cool that it's based on CSS variables for Global styles support, this has the downside of potentially requiring some CSS tweak from theme authors in order to have less specificity for their line-height styles to avoid overriding the CSS variable. (I think this work is needed regardless of this PR)
 - Right now, the biggest downside is that we end up with two panels on the sidebar "Typography" and "Text Settings", I believe that's essentially because the remaining paragraph customizations (drop cap, font-sizes) might need to be refactored to rely on the CSS vars implementation too. An alternative in the meantime is possible by introducing Slots (like we do for the advanced section) but maybe the double pannel is fine for now.

I'd appreciate some thoughts on the approach and if you like it, feel free to include in this PR.

Reply:
@youknowriad I'm in favour of this approach! It feels more inline with colors and global styles.

> Right now, the biggest downside is that we end up with two panels on the sidebar "Typography" and "Text Settings"

Yes... that's a bit of a concern 😅 

In theory, once more (style) attributes move over to this new system, it'll be smoothed out. This may be the first of many in that migration.

The only issue I can see is that the base style isn't as specific enough to override the styles from themes (or even the editor).

I'll merge your updates in https://github.com/WordPress/gutenberg/commit/6c7dcbaa9458659254c7d169cf791918bf0dd9eb into this PR